benchmark.py

```
@@ -0,0 +1,833 @@
+        self.process = subprocess.Popen(
+            command,
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.DEVNULL,
+            stderr=self.log_file.open('a'),
+        )
+        await self.wait_for_ready()
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        if self.process is not None:
+            if self.process.returncode is None:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)